### ISSUE TITLE
Weapons should continue swinging/firing when mouse is held down

### ISSUE THREAD

**Issue by sowelipililimute** [space-wizards/space-station-14#42573]
This is a basic part of accessibility in how users interact with weapons. UI should not encourage repetitive inputs where it can be avoided.

**Comment by ScarKy0** [comment]
Doesnt this kinda beat the point of automatic guns?

**Comment by EthanQix** [comment]
> Doesnt this kinda beat the point of automatic guns?

Is there any gameplay reason to differentiate automatic from non-automatic weapons ?

**Comment by PJB3005** [comment]
> Doesnt this kinda beat the point of automatic guns?

If the point of automatic guns is to avoid getting repetitive strain injury, automatic guns should not exist.

**Comment by SlamBamActionman** [comment]
There is an easy way topple me off any hill I'm dying on:

<img width="598" height="398" alt="Image" src="https://github.com/user-attachments/assets/2e2e801e-7497-4135-88f6-86966604873c" />

I'm 100% for this. Semi-auto guns/melee in this game is not so important as to define its combat and I will gladly sacrifice it in favor of reducing RSI issues for those it impacts.

**Comment by deathride58** [comment]
A lot of games as of late have been adopting automatic firing for semi-auto weaponry. Hell, semi-auto weapons having auto-fire even dates back to the 90's with Half-Life 1 and Doom both featuring it, and even in the 2000's it's been present in Halo and Half-Life 2.

Usually the way games balance it is by either keeping a consistent max fire-rate when autofiring semi-auto weapons (like in Half-Life 2 where the pistol can fire literally as fast as you can click, but autofire shoots at around 130 RPM), or introducing an accuracy penalty (like in Halo 1 and Doom, where the pistols don't fire any faster when you autofire, but they lose accuracy if you either hold down the trigger or buffer your input).

However, there's also games like Warframe showing that it can be objectively fine to have it be no different from clicking manually (In Warframe it's balanced by most semi-auto weapons having an ungodly amount of recoil, so if you shove a bunch of firespeed mods onto your rubico prime you're gonna have to be very good at recoil control to actually tame it when you let it rip)

Looking at SS14, the vast majority of weapons in the game have fairly slow firerates to begin with (Most SMGs sit well under 600 RPM, for instance), and semi-auto weapons in particular are already capped slow enough that they wouldn't be out of line if they were full-auto (the fastest pistol caps out at 360 RPM).

This leaves us with the feeling that adding autofire would have zero impact on the actual meta. The only difference would be that those with RSIs (like us!) no longer have to put up with physical pain just to use a semi-auto weapon

**Comment by PJB3005** [comment]
> (like in Half-Life 2 where the pistol can fire literally as fast as you can click, but autofire shoots at around 130 RPM)

Should be noted that TF2 also did this initially and Valve got rid of it at some point. I wouldn't take this as a good thing to do because that just means people giving themselves RSI are advantaged.

**Comment by ThatGuyUSA** [comment]
There's something that defines this function somewhere in the code.

Drills, northstars, and chainsaws use it for melee options!

Now maybe it should be an accessibility feature that can be toggle through the settings? Or maybe just add it to the base weapon. 

I dunno, not many downsides to adding this other than making the throngler stronger which isn't a real problem. 

As for guns, well there's already auto shooting guns, and I can't think of when holding down the fire button for a shotgun/hristov/other burst damage weapon would be better than just timing your shots.

**Comment by deltanedas** [comment]
throngler already autoswings

**Comment by SnappingOpossum** [comment]
Throngler doesn't actually, crusher dagger and  are more that do though.
